src/activitys/ChallengeModeDungeon.ts

```diff
@@ -0,0 +1,158 @@
+import { Metadata } from "main/types";
+import { dungeonTimersByMapId, VideoCategory } from "../main/constants";
+import { ChallengeModeTimelineSegment, TimelineSegmentType } from "../main/keystone";
+import Activity from "./Activity";
+
+export default class ChallengeModeDungeon extends Activity {
+    private mapID: number;
+    private level: number;
+    private affixes: number[];
+    private timings: number[];
+    private CMDuration: number = 0;
+    private chests?: number;
+    private timelineSegments: ChallengeModeTimelineSegment[] = []
+
+    constructor(startDate: Date, 
+                zoneID: number, 
+                mapID: number, 
+                level: number, 
+                affixes: number[])
+    {
+        super(startDate, VideoCategory.MythicPlus);
+        this.zoneID = zoneID;
+        this.mapID = mapID;
+        this.level = level;
+        this.affixes = affixes; 
+        this.timings = dungeonTimersByMapId[mapID];
+    }
+
+    endChallengeMode(endDate: Date, CMDuration: number) {
+        this.endCurrentTimelineSegment(endDate);
+
+        // If last timeline segment is less than 10 seconds long, discard it.
+        // It's probably not useful
+        const lastTimelineSegment = this.getCurrentTimelineSegment();
+
+        if (lastTimelineSegment && lastTimelineSegment.length() < 10000) {
+            console.debug("[ChallengeModeDungeon] Removing last timeline segment because it's too short.");
+            this.removeLastTimelineSegment();
+        }
+
+        this.endDate = endDate;
+        this.CMDuration = CMDuration;
+        this.result = true; // @@@ is this OK or need to handle non completed better? 
```

Comment:
Basically:

```ts
this.result = this.chests > 0
```

Reply:
Not sure I'm with you here... it's possible to complete a key with 0 upgradeLevel right?

Reply:
i.e. depleted but still killed all the bosses

Reply:
Yes, you're right - I misunderstood the intention - `this.result` replaces the old `Metadata.result` and `this.chests` is basically equivalent to the old `ChallengeModeDungeon.result`. No problem then :)